### PR TITLE
Automate Workflow to Production

### PR DESCRIPTION
**Pull request by sfawcett123**
### Context
The users require a workflow which delivers change to a review, and then once reviewed, fully tests the code and delivers it to production.

### Changes
The build workflow triggers, from either a **PR** or a **Push**.
  If it is a **push** it will build a docker image, test it and deliver it to a Review Instance.
  When the PR is merged it will destroy the Review Instance, rebuild the merged code into a new docker image, carry out all the tests, and deliver it to Development. If that is successful, it will deliver it to the QA space, where the automated cypress tests will be run against it. After that, a release will be created and the image will be delivered to production.

There is a manual workflow, which permits the release of a 'Release' to a selected environment

### Review
The review is complicated by the github workflows needing to be run in github, and are relying on other workflows, which may not exist in the master branch. A lot of the processes are copied from the TTA workflow, which works.

### CODE REVIEW

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -113,84 +104,290 @@ jobs:
...
-        if: always()
-        uses: bobheadxi/deployments@v0.4.2
-        with:
-          step: finish
-          token: ${{ secrets.GITHUB_TOKEN}}
-          env: 'Development'
-          status: ${{ job.status }}
-          deployment_id: ${{ steps.deployment.outputs.deployment_id }}
-          ref: ${{ github.sha }}
+      - name: Post PR comment
```
Comment (saliceti) [inline .github/workflows/build.yml:157]: Can we try https://github.com/marketplace/actions/sticky-pull-request-comment instead?

File: .github/workflows/cve.yml [hunk 2]
```diff
@@ -1,29 +1,39 @@
...
   DOMAIN: london.cloudapps.digital
+  PROJECT: Get Into Teaching Application
+  REVIEW_APPLICATION: review-get-into-teaching-app
+  APPLICATION_NAME:   get-into-teaching-app
 
 jobs:
-  owasp:
-    name: OWASP Test
+  cve:
+    name: 'CVE Test'
```
Comment (saliceti) [inline .github/workflows/cve.yml]: Remove quotes please 😀

File: .github/workflows/build.yml (hunk 1 above)
Comment (sfawcett123) [reply .github/workflows/build.yml:157]: yup

File: .github/workflows/cve.yml (hunk 2 above)
Comment (sfawcett123) [reply .github/workflows/cve.yml]: done

File: .github/workflows/deploy.yml [hunk 3]
```diff
@@ -0,0 +1,178 @@
...
+             TF_VAR_password:          "${{ secrets.GOVUKPAAS_PASSWORD  }}"
+             TF_VAR_RAILS_ENV:         "${{ steps.variables.outputs.RAILS_ENV  }}"
+             TF_VAR_paas_adviser_docker_image: ${{env.DOCKERHUB_REPOSITORY}}:review-${{ steps.sha.outputs.short}} 
+             TF_VAR_RAILS_MASTER_KEY:  "${{ steps.variables.outputs.RAILS_MASTER_KEY  }}"
+             TF_VAR_HTTPAUTH_PASSWORD: "${{ steps.variables.outputs.HTTP_PASSWORD  }}"
+             TF_VAR_HTTPAUTH_USERNAME: "${{ steps.variables.outputs.HTTP_USERNAME  }}"
+             TF_VAR_docker_username:   "${{ secrets.DOCKERHUB_USERNAME }}"
+             TF_VAR_docker_password:   "${{ secrets.DOCKERHUB_PERSONAL_ACCESS_TOKEN }}"
+
+       - name: Terraform (none review)
```
Comment (saliceti) [inline .github/workflows/deploy.yml]: "non review" ?

Comment (sfawcett123) [reply .github/workflows/deploy.yml]: replaced with **Terraform ( ${{github.event.inputs.environment}} )**

**Review by saliceti — APPROVED** [review]

**Review by vigneshmsft — APPROVED** [review]
👏🏽

**Pull request merged**